caddytls/setup.go

```diff
@@ -164,6 +164,9 @@ func setupTLS(c *caddy.Controller) error {
 			case "max_certs":
 				c.Args(&maxCerts)
 				config.OnDemand = true
+			case "ask":
+				c.Args(&askURL)
```

Comment:
See if you can parse the URL here, and if it fails, return an error. This will help prevent configuration errors after the server has already started...

Reply:
See line 220 in the latest revision which validates the URL can be parsed and the scheme is either HTTP or HTTPS.